src/arkode/arkode.c

```diff
@@ -1969,17 +2040,30 @@ int arkInitialSetup(ARKodeMem ark_mem, sunrealtype tout)
     /* Estimate initial h if not set */
     if (ark_mem->h == ZERO)
     {
-      /* Again, temporarily set h for estimating an optimal value */
+      /* If necessary, temporarily set h as it is used to compute the tolerance
+         in a potential mass matrix solve when computing the full rhs */
       ark_mem->h = SUNRabs(tout - ark_mem->tcur);
-      if (ark_mem->h == ZERO) { ark_mem->h = ONE; }
+      if (ark_mem->h == ZERO) ark_mem->h = ONE;
+
+      /* Call fullrhs if needed */
+      if (!(ark_mem->fn_is_current))
+      {
+        retval = ark_mem->step_fullrhs(ark_mem, ark_mem->tcur, ark_mem->yn,
+                                       ark_mem->fn, ARK_FULLRHS_START);
+        if (retval != 0) { return (ARK_RHSFUNC_FAIL); }
+        ark_mem->fn_is_current = SUNTRUE;
+      }
+
       /* Estimate the first step size */
       tout_hin = tout;
       if (ark_mem->tstopset &&
           (tout - ark_mem->tcur) * (tout - ark_mem->tstop) > ZERO)
       {
         tout_hin = ark_mem->tstop;
       }
-      hflag = arkHin(ark_mem, tout_hin);
+      hflag = arkHin(ark_mem, ark_mem->tcur, tout_hin, ark_mem->yn, ark_mem->fn,
+                     ark_mem->ycur, ark_mem->tempv1, ark_mem->tempv2,
+                     ark_mem->step_fullrhs, &ark_mem->h);
```

Comment:
Again, it has been over a year since I did this piece, so my recollection is rusty.  I believe that the point was to be able to specify the vectors that `arkHin` would use and fill in the estimation process, instead of requiring those to be hard-coded shortcuts inside the function.  Before we decided that the $H$-$h$ adaptivity approach needed to be removed, `mriStep_Init` would call **both** `mriStep_Hin` *and* `arkHin` so that it could set the initial values of both $H_0$ and $h_0$.  After removing $H$-$h$ adaptivity, that necessity is now alleviated, so the changes to this function could be reverted; however, I still like that in its current form, `arkHin` could be called from more "users" than just `arkInitialSetup`.

Reply:
I'll leave this and the preceding comment about `arkHin` as unresolved.  However, as I mentioned in my last comment I prefer the updated approach, so if others feel strongly otherwise then we can discuss.  Alternately, @gardner48 and/or @Steven-Roberts can "resolve" these.